backend/src/main/java/de/neuefische/teddyscoronadiaries/service/WeatherService.java

```diff
@@ -37,35 +32,24 @@ public ProvinceCapitalWeatherData getProvinceCapitalWeatherData(String province)
             throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Province not known");
         }
 
-        String capitalWoeid = provinceData.get().getCapitalWoeid();
-        Optional<WeatherData> weatherDataForProvinceCapital = getWeatherDataForProvinceCapital(capitalWoeid);
-
-        if(weatherDataForProvinceCapital.isEmpty()) {throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Weather API not available");}
-
-        return getProvinceCapitalWeatherDataFromWeatherData(provinceData.get().getCapital(), weatherDataForProvinceCapital.get());
-    }
-
-    public Optional<WeatherData> getWeatherDataForProvinceCapital(String capitalWoeid) {
-        List<WeatherData> weatherData = metaWeatherApiService.getWeatherData(capitalWoeid);
+        String capital = provinceData.get().getCapital();
+        Optional<Weather> weatherData = openWeatherApiService.getWeatherForProvinceCapital(capital);
 
-        String currentDay = currentDateUtils.getCurrentDay();
+        if(weatherData.isEmpty()) {throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Weather API not available");}
 
-        return weatherData.stream().filter(dataSet -> dataSet.getApplicableDate().equals(currentDay)).findAny();
+        return getProvinceCapitalWeatherDataFromWeatherData(provinceData.get().getCapital(), weatherData.get());
     }
 
-    public ProvinceCapitalWeatherData getProvinceCapitalWeatherDataFromWeatherData(String capital, WeatherData data) {
+    public ProvinceCapitalWeatherData getProvinceCapitalWeatherDataFromWeatherData(String capital, Weather data) {
 
-        if(WeatherCategories.getWeatherCategory(data.getWeatherStateAbbreviation().toUpperCase()).isEmpty()) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Unknown weather category");
-        }
 
         return new ProvinceCapitalWeatherData(
                 capital,
-                WeatherCategories.getWeatherCategory(data.getWeatherStateAbbreviation().toUpperCase()).get(),
-                (int)Math.round(data.getMinTemperature()),
-                (int)Math.round(data.getMaxTemperature()),
-                (int)Math.round(data.getCurrentTemperature()),
-                "https://www.metaweather.com/static/img/weather/" + data.getWeatherStateAbbreviation() + ".svg"
+                data.getWeather().get(0).getWeatherDescription(),
+                (int)Math.round(data.getTemperature().getMinTemperature()),
+                (int)Math.round(data.getTemperature().getMaxTemperature()),
+                (int)Math.round(data.getTemperature().getCurrentTemperature()),
```

Comment:
Why not return a rounded double and not cast here? Also using a builder safes me from thinking what each value means.

Reply:
don't wanna have to round / trim / whatever in the frontend so I do it here ... also I do not want to use a builder here to ensure that all values are defined for the frontend 😝

Reply:
Yeah, but you have a test for that ;)